_posts/2017-08-18-Refactoring-with-reactJS-shared-state.md

```diff
@@ -0,0 +1,252 @@
+---
+layout: post
+author: aliOrlando
+date: 2017-08-18 13:30:00 -0500
+title: "Shared State: Refactoring a Legacy Application with React - Part 2"
+excerpt: "When we started on this whole project of finding a pattern to refactor in React, we knew we wanted to keep the effort small. Everyone was on the same page: let’s get React in and get it rendering something."
+---
+*Note: This post is part of a series, starting with [Refactoring a Legacy Application with React - Introduction]({% post_url 2017-08-14-Refactoring-with-reactJS-introduction %})*
+
+When we started on this whole project of finding a pattern to refactor in React, we knew we wanted to keep the effort small. Everyone was on the same page: let’s get React in and get it rendering something. If you’ve read Part 1, then you know that we got there. We have React components rendered into the page. Yay! Problem solved!
+
+As you probably know, solving a problem can have a way of creating more problems that you hadn’t thought about before. So what hadn’t we thought about before?
+
+State management.
```

Comment:
And I would also mention that our goal is to refactor multiple areas dispersed on the page that would not be nested into a single container component. Thus we would not have means of exchanging information between those components but through a common global state management system.